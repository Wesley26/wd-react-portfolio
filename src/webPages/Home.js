import React from 'react'
import { noImage,
        websiteMasterLogo,
        philly_P,
        philly_L,
        homeText } from './contentMaster/PageContents.js'

function Home(){
    return (
        <div className="font-body bg-gray-100 m-6 shadow-xl">

            <div className="p-6">
                <div className="pb-10">
                    <div className="flex justify-center items-center">
                        <img
                        src={websiteMasterLogo}
                        alt={noImage}
                        className="focus:outline-none">
                        </img>
                    </div>
                    <h3 className="font-title text-xl text-center font-bold">
                        {homeText.subHeaderH}
                    </h3>
                    <br>
                    </br>
                    <p className="text-lg">
                        {homeText.mainParagraphH}
                    </p>
                    <br>
                    </br>
                    <div>
                        <figure>
                            <img 
                                src={philly_P} 
                                alt={noImage}
                                className="lg:hidden focus:outline-none shadow-lg">
                            </img>
                            <img 
                                src={philly_L} 
                                alt={noImage}
                                className="invisible h-1 w-1 lg:visible lg:h-full lg:w-full focus:outline-none shadow-lg">
                            </img>
                            <figcaption className="pt-2 text-left">
                                {homeText.captionH}
                            </figcaption>
                        </figure>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Home