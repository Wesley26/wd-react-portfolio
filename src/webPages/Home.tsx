import { FC } from 'react';
import PageContents from './contentMaster/PageContents';

const Home:FC = () => {

    const pageContents = PageContents();

    return (
        <div className="font-body bg-body-lightGray m-6 shadow-xl">

            <div className="p-6">
                <div className="pb-10">
                    <div className="flex justify-center items-center">
                        <img
                        src={pageContents.allImages.masterLogo}
                        alt={pageContents.noImage}
                        className="focus:outline-none">
                        </img>
                    </div>
                    <h3 className="font-title text-xl text-center font-bold">
                        {pageContents.homeText.subHeaderH}
                    </h3>
                    <br>
                    </br>
                    <p className="text-lg">
                        {pageContents.homeText.mainParagraphH}
                    </p>

                    <br>
                    </br>

                    <div>
                        <figure>
                            <img 
                                src={pageContents.allImages.phillyPort} 
                                alt={pageContents.noImage}
                                className="lg:hidden focus:outline-none shadow-lg">
                            </img>
                            <img 
                                src={pageContents.allImages.phillyLand} 
                                alt={pageContents.noImage}
                                className="invisible h-1 w-1 lg:visible lg:h-full lg:w-full focus:outline-none shadow-lg">
                            </img>
                            <figcaption className="pt-2 text-left">
                                {pageContents.homeText.captionH}
                            </figcaption>
                        </figure>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Home