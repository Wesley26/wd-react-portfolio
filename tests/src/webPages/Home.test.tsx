import { render } from '@testing-library/react';
import Home from '../../../src/webPages/Home';
import '../../src/index.css';

it(`render App.tsx test component`, () => {

    const view = (render(<Home />));

    expect(view).toMatchSnapshot();

});