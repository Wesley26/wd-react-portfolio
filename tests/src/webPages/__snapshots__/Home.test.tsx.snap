// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App.tsx test component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="font-body bg-body_lightGray m-6 shadow-xl"
      >
         
        <div
          class="p-6"
          style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
        >
          <div
            class="pb-10"
          >
            <div
              class="flex justify-center items-center"
            >
              <img
                alt="Image has failed to load."
                class="focus:outline-none"
                src="[object Object]"
              />
            </div>
            <h3
              class="font-title text-xl text-center font-bold"
            >
              Web and Mobile Design: Perfectly Balanced To Meet Your Future Business Image
            </h3>
            <br />
            <p
              class="text-lg"
            >
              Are you searching for a new, high-tech, scalable web and/or mobile design to welcome your customers?
                            Do you have software engineers on your current team who specialize behind the scenes, but do 
                            not work as often on what the end-user views? I can help you! Feel free to browse my portfolio 
                            gallery for all the examples of web and mobile design I can provide to your software engineer 
                            team! Use the menu button on the top right of your screen to begin.
            </p>
            <br />
            <div>
              <figure>
                <img
                  alt="Image has failed to load."
                  class="lg:hidden focus:outline-none shadow-lg"
                  src="[object Object]"
                />
                <img
                  alt="Image has failed to load."
                  class="invisible h-1 w-1 lg:visible lg:h-full lg:w-full focus:outline-none shadow-lg"
                  src="[object Object]"
                />
                <figcaption
                  class="pt-2 text-left"
                >
                  The photo above was taken in person outside the South-East doorway of the Philadelphia Art Museum overlooking 
                                the famed Rocky Steps in Philadelphia, P.A.
                </figcaption>
              </figure>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="font-body bg-body_lightGray m-6 shadow-xl"
    >
       
      <div
        class="p-6"
        style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
      >
        <div
          class="pb-10"
        >
          <div
            class="flex justify-center items-center"
          >
            <img
              alt="Image has failed to load."
              class="focus:outline-none"
              src="[object Object]"
            />
          </div>
          <h3
            class="font-title text-xl text-center font-bold"
          >
            Web and Mobile Design: Perfectly Balanced To Meet Your Future Business Image
          </h3>
          <br />
          <p
            class="text-lg"
          >
            Are you searching for a new, high-tech, scalable web and/or mobile design to welcome your customers?
                            Do you have software engineers on your current team who specialize behind the scenes, but do 
                            not work as often on what the end-user views? I can help you! Feel free to browse my portfolio 
                            gallery for all the examples of web and mobile design I can provide to your software engineer 
                            team! Use the menu button on the top right of your screen to begin.
          </p>
          <br />
          <div>
            <figure>
              <img
                alt="Image has failed to load."
                class="lg:hidden focus:outline-none shadow-lg"
                src="[object Object]"
              />
              <img
                alt="Image has failed to load."
                class="invisible h-1 w-1 lg:visible lg:h-full lg:w-full focus:outline-none shadow-lg"
                src="[object Object]"
              />
              <figcaption
                class="pt-2 text-left"
              >
                The photo above was taken in person outside the South-East doorway of the Philadelphia Art Museum overlooking 
                                the famed Rocky Steps in Philadelphia, P.A.
              </figcaption>
            </figure>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
